Add plugin: Yearly Glance

Moyf opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/Moyf/yearly-glance

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
["description": "Year at a Glance - Overview of Annual Events with Customizable Management Options.",](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/manifest.json#L6), [name: "View Type",](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/i18n/locales/en.ts#L19), [name: "Show Weekdays",](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/i18n/locales/en.ts#L27), [name: "Highlight Today",](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/i18n/locales/en.ts#L31), [name: "Highlight Weekends",](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/i18n/locales/en.ts#L35), many others
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[console.debug("[yearly-glance] 加载年度概览插件");](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/main.ts#L34), [console.debug("[yearly-glance] 卸载年度概览插件");](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/main.ts#L57), [console.debug(\`[yearly-glance] 初始化节日数组\`);](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/core/utils/dataMerge.ts#L42), [console.debug(\`[yearly-glance] 重置非数组类型的节日数据为空数组\`);](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/core/utils/dataMerge.ts#L48), [console.debug(](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/core/utils/dataMerge.ts#L94-L96), [console.debug(](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/core/utils/dataMerge.ts#L102-L106), [console.debug(](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/core/utils/dataMerge.ts#L109-L111), [console.debug(](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/core/utils/dataMerge.ts#L131-L133)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[const lang = window.localStorage.getItem("language") || "en";](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/i18n/i18n.ts#L16)
Please use [getLanguage](https://docs.obsidian.md/Reference/TypeScript+API/getLanguage) to get the user's current language. Note that you'll also need to update your [minAppVersion](https://docs.obsidian.md/Reference/TypeScript+API/PluginManifest/minAppVersion) to `1.8.0`.

[\<h2>{t("setting.title")}<\/h2>](https://github.com/Moyf/yearly-glance/blob/5eb5e94c1f2cd15346e711084d3e94910f5f145d/src/components/Settings/SettingsTab.tsx#L45)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

Moyf commented:
Hi @Zachatoo , we've made the changes in [here](https://github.com/Moyf/yearly-glance/pull/42/files)
Thanks for your reviews 🫡 

> For the `console.debug()` functions, we disabled it now, so the users shall not see any of them:
> ![image](https://github.com/user-attachments/assets/686d0df0-2a8c-4bfb-927f-48a046383f24)

RavenHogWarts commented:
We have implemented the following changes to ensure logging occurs only during development:

esbuild.config.mjs[#L16](https://github.com/Moyf/yearly-glance/blob/4c793b6b56bc88b066f841444fdecd8e47e19362/scripts/esbuild.config.mjs#L16) [#L120](https://github.com/Moyf/yearly-glance/blob/4c793b6b56bc88b066f841444fdecd8e47e19362/scripts/esbuild.config.mjs#L120)

package.json[#L8-L9](https://github.com/Moyf/yearly-glance/blob/4c793b6b56bc88b066f841444fdecd8e47e19362/package.json#L8-L9)

Moyf commented:
@Zachatoo @ObsidianReviewBot Excuse me, may I ask if I need to do anything else to update the status of this PR, or is it just a matter of waiting?

Zachatoo commented:
The bot didn't see that you changed the `console.debug` lines, so it thinks you haven't made the requested changes. I can move this forward manually.

joethei commented:
["description": "Year at a Glance - Overview of Annual Events with Customizable Management Options.",](https://github.com/Moyf/yearly-glance/blob/cdb5b6b54675b2e72a8ffe06d566c444ebebe9b1/manifest.json#L6)
This should use sentence case.

[name: "General",](https://github.com/Moyf/yearly-glance/blob/cdb5b6b54675b2e72a8ffe06d566c444ebebe9b1/src/i18n/locales/en.ts#L8)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[desc: "View configuration",](https://github.com/Moyf/yearly-glance/blob/cdb5b6b54675b2e72a8ffe06d566c444ebebe9b1/src/i18n/locales/en.ts#L9)
[Avoid "configuration" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

Moyf commented:
> ["description": "Year at a Glance - Overview of Annual Events with Customizable Management Options.",](https://github.com/Moyf/yearly-glance/blob/cdb5b6b54675b2e72a8ffe06d566c444ebebe9b1/manifest.json#L6) This should use sentence case.
> 
> [name: "General",](https://github.com/Moyf/yearly-glance/blob/cdb5b6b54675b2e72a8ffe06d566c444ebebe9b1/src/i18n/locales/en.ts#L8) [Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
> 
> [desc: "View configuration",](https://github.com/Moyf/yearly-glance/blob/cdb5b6b54675b2e72a8ffe06d566c444ebebe9b1/src/i18n/locales/en.ts#L9) [Avoid "configuration" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

Hello, we've made the changes and released a new version (`2.4.0`)
I've also changed the description in `community-plugins.json` to follow the sentence case rule.

Additionally, I would like to point out that our base class requires mandatory name and description attributes, so we have retained the localized text for "general." In fact, we have already removed the top-level title structure in [this commit](https://github.com/Moyf/yearly-glance/commit/54cc5ee042e42f11908bcfc3afaf4d6830065e53#diff-44eb41c6f5835d7c8b93c24567a76f452dc05ad3e2ea291c4ed7762ef9cca590L52).
